### PR TITLE
Add support for "clib i"

### PR DESCRIPTION
**Pull request by Aslemammad**
#186 Feature: Done.😍
I'll add more short commands in the future. ♥️

### CODE REVIEW

File: src/clib.c [hunk 1]
```diff
@@ -20,32 +20,34 @@
...
+    "\n"
+    "  Options:\n"
+    "\n"
+    "    -h, --help     Output this message\n"
+    "    -v, --version  Output version information\n"
+    "\n"
+    "  Commands:\n"
+    "\n"
+    "    init               Start a new project\n"
+    "    install [name...], i [name...]  Install one or more packages\n"
```
Comment (jwerle) [inline src/clib.c:34]: Can you make the descriptions appear in the same column for each line?

File: src/clib.c [hunk 2]
```diff
@@ -20,32 +20,34 @@
...
+    "    search [query]     Search for packages\n"
+    "    help <cmd>         Display help for cmd\n"
+    "";
+
+#define format(...) ({                              \
+  if (-1 == asprintf(__VA_ARGS__))                  \
```
Comment (jwerle) [inline src/clib.c:40]: No newlines before `{`

File: src/clib.c [hunk 3]
```diff
@@ -20,32 +20,34 @@
...
+    rc = 1;                                         \
+    fprintf(stderr, "Memory allocation failure\n"); \
+    goto cleanup;                                   \
+  }                                                 \
 })
 
-int
-main(int argc, const char **argv) {
+int main(int argc, const char **argv)
+{
```
Comment (jwerle) [inline src/clib.c:49]: Same here

File: src/clib.c [hunk 4]
```diff
@@ -109,28 +121,33 @@ main(int argc, const char **argv) {
...
     format(&command_with_args, "%s", bin);
   }
 
   debug(&debugger, "exec: %s", command_with_args);
 
   rc = system(command_with_args);
   debug(&debugger, "returned %d", rc);
-  if (rc > 255) rc = 1;
-
+  if (rc > 255)
```
Comment (jwerle) [inline src/clib.c:149]: This should've had an enclosing `{ rc = 1; }`

**Comment by jwerle** [comment]
Other than the small style changes, this LGTM

File: src/clib.c [hunk 5]
```diff
@@ -56,51 +58,61 @@ main(int argc, const char **argv) {
...
+    {
       args = str_flatten(argv, 2, argc);
-      if (NULL == args) goto cleanup;
+      if (NULL == args)
+        goto cleanup;
     }
   }
   debug(&debugger, "args: %s", args);
 
+  cmd = strcmp(cmd, "i") == 0 ? strdup("install") : cmd; // equality of "i" and "install" like npm
```
Comment (stephenmathieson) [inline src/clib.c:115]: I don't think we need to allocate memory for a static string. Can we work around the `strdup("install")` here?

**Comment by Aslemammad** [comment]
ok, I'll solve them

File: src/clib.c (hunk 5 above)
Comment (Aslemammad) [reply src/clib.c:115]: But, it will have an error. `munmap_chunk(): invalid pointer`.
If i don't put strdup.

**Comment by Aslemammad** [comment]
I think I solved the problems, any opinion?

**Comment by Aslemammad** [comment]
😍

**Comment by Aslemammad** [comment]
I appreciate this, Thank you so much.

**Pull request merged**